### PR TITLE
feat(migrations): migrate existing rollapps plus several minors

### LINKED ISSUE #1411: migration: rollapp and sequencer meta data
See https://docs.google.com/spreadsheets/d/1wGMQbrsGAwiSAmRGHbM8oDAkklH1lHqgcdxU3YsMIYw/edit?gid=394826780#gid=394826780

(Note: also contains reward address migration to avoid toe stepping)

### PR DESCRIPTION
**Pull request by keruch**
## Description

All issues from #1411 and #1409 

----

Closes #1411 #1409 

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: app/upgrades/v4/upgrade.go [hunk 1]
```diff
@@ -78,6 +78,14 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		if err := migrateRollappRegisteredDenoms(ctx, keepers.RollappKeeper); err != nil {
+			return nil, err
+		}
+
+		if err := migrateDymnsParams(ctx, keepers.DymNSKeeper); err != nil {
```
Comment (omritoptix) [inline app/upgrades/v4/upgrade.go:85]: I don't think we need to migrate dymns params. just add the dymns to the new modules list in the `Added` of the create upgrade handler

Comment (keruch) [reply app/upgrades/v4/upgrade.go]: done! plus also added sponsorship and iro modules

**Review by omritoptix — APPROVED** [review]

File: app/upgrades/v4/upgrade.go (hunk 1 above)
Comment (omritoptix) [reply app/upgrades/v4/upgrade.go]: nice. I think we're missing `grouptypes`

File: app/upgrades/v4/constants.go [hunk 2]
```diff
@@ -20,6 +24,9 @@ var Upgrade = upgrades.Upgrade{
 			consensustypes.ModuleName,
 			crisistypes.ModuleName,
 			lightclienttypes.ModuleName,
+			sponsorshiptypes.ModuleName,
+			dymnstypes.ModuleName,
+			irotypes.ModuleName,
```
Comment (omritoptix) [inline app/upgrades/v4/constants.go:29]: I believe we're missing `grouptypes.ModuleName`

**Review by omritoptix — CHANGES_REQUESTED** [review]

**Review by danwt — APPROVED** [review]
gj, just Omri's comment to fix

File: app/upgrades/v4/constants.go (hunk 2 above)
Comment (keruch) [reply app/upgrades/v4/constants.go:30]: added

**Review by omritoptix — APPROVED** [review]

**Pull request merged**